hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -270,6 +275,10 @@ public V get(K key) {
     public void put(K key, V value) {
         checkAvailable();
 
+        if (!isEvictionEnabled() && maxSizeChecker != null && maxSizeChecker.isReachedToMaxSize()) {
+            return;
```

Comment:
We just discussed this. Of course, with an _existing_ key we'll produce a stale value here. Looks like a bug in the other implementations though, which I copied.

That's a very good finding!

I'll write a test for this and fix this.